src/jrd/fun.epp

```diff
@@ -423,14 +425,10 @@ void FUN_evaluate(thread_db* tdbb, const Function* function, const NestValueArra
 			// doesn't carry the null flag in the descriptor. Why Borland didn't
 			// set on such flag is maybe because it only has local meaning.
 			// This closes SF Bug #728839.
-			if ((request->req_flags & req_null) && !(input && (input->dsc_flags & DSC_null)))
-			{
-			    *arg_ptr++ = NULL;
-			}
+			if (!input && !(input && (input->dsc_flags & DSC_null)))
```

Comment:
Condition should be either (!input) if we want to return DSC_null "as is", or something like this:
if (!input || (input->dsc_flags & DSC_null)) if we want to transform DSC_null into nullptr inside *arg_ptr.